ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationPool.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import tech.pegasys.teku.core.operationvalidators.OperationStateTransitionValidator;
+import tech.pegasys.teku.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.datastructures.operations.ProposerSlashing;
+import tech.pegasys.teku.datastructures.operations.SignedVoluntaryExit;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+import tech.pegasys.teku.ssz.SSZTypes.SSZList;
+import tech.pegasys.teku.ssz.SSZTypes.SSZMutableList;
+import tech.pegasys.teku.util.config.Constants;
+
+public class OperationPool<T> {
+
+  private static Map<Class<?>, Integer> maxNumberOfElementsInBlock =
+      Map.of(
+          SignedVoluntaryExit.class, Constants.MAX_VOLUNTARY_EXITS,
+          ProposerSlashing.class, Constants.MAX_PROPOSER_SLASHINGS,
+          AttesterSlashing.class, Constants.MAX_ATTESTER_SLASHINGS);
+
+  private Set<T> operations = new HashSet<>();
```

Comment:
We probably want to limit the size of this right?  
```suggestion
  private Set<T> operations = LimitedSet.create(...);
```

Reply:
Yeah we probably do. Any suggestions on the sizes of these sets?

Reply:
Done. Put 1000 for now.